src/features/navigation/components/SubformsForPage.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+
+import { ChevronDownIcon } from '@navikt/aksel-icons';
+import cn from 'classnames';
+
+import { ContextNotProvided } from 'src/core/contexts/context';
+import { useIsProcessing } from 'src/core/contexts/processingContext';
+import { useDataTypeFromLayoutSet } from 'src/features/form/layout/LayoutsContext';
+import { useStrictDataElements } from 'src/features/instance/InstanceContext';
+import { Lang } from 'src/features/language/Lang';
+import classes from 'src/features/navigation/components/SubformsForPage.module.css';
+import { useNavigate, useNavigationParams } from 'src/features/routing/AppRoutingContext';
+import { isSubformValidation } from 'src/features/validation';
+import { useComponentValidationsForNode } from 'src/features/validation/selectors/componentValidationsForNode';
+import {
+  getSubformEntryDisplayName,
+  useExpressionDataSourcesForSubform,
+  useSubformFormData,
+} from 'src/layout/Subform/utils';
+import { NodesInternal, useNode } from 'src/utils/layout/NodesContext';
+import { useNodeItem } from 'src/utils/layout/useNodeItem';
+import type { ExprVal, ExprValToActualOrExpr } from 'src/features/expressions/types';
+import type { IData } from 'src/types/shared';
+
+export function SubformsForPage({ pageKey }: { pageKey: string }) {
+  const subformIds = NodesInternal.useLaxMemoSelector(({ nodeData }) =>
+    Object.values(nodeData)
+      .filter((node) => node.pageKey === pageKey && node.layout.type === 'Subform')
+      .map((node) => node.layout.id),
+  );
+
+  if (subformIds === ContextNotProvided || !subformIds.length) {
+    return null;
+  }
+
+  return subformIds.map((nodeId) => (
+    <SubformGroup
+      key={nodeId}
+      nodeId={nodeId}
+    />
+  ));
+}
+
+function SubformGroup({ nodeId }: { nodeId: string }) {
+  const [isOpen, setIsOpen] = useState(false);
+  const node = useNode(nodeId);
+  if (!node?.isType('Subform')) {
+    // This should never happen, @see SubformsForPage
+    throw new Error(`Navigation expected component: "${nodeId}" to exist and be of type: "Subform"`);
+  }
+  const subformIdsWithError = useComponentValidationsForNode(node).find(isSubformValidation)?.subformDataElementIds;
```

Comment:
Is there always just one validation with a list of ids? :thinking: I haven't looked into this deeply, but I would assume there could be other validations skipped here because you're using `.find()` instead of `.filter()`.

Reply:
Yeah its just one, and it says something like "There are errors in one or more subform entries", and the `subformDataElementIds` allow us to highlight which rows contain errors. I think it was done this way to not spam lots of validation errors that would look pretty much the same for each row. Though, now that we have `entryDisplayName`, we could show a unique error message per row, but previously they would either look exactly the same or include the uuid, which was not desirable.